components/compositing/compositor.rs

```diff
@@ -1951,14 +2210,14 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
             },
         };
 
-        // Nottify embedder that servo is ready to present.
+        // Notify embedder that servo is ready to present.
         // Embedder should call `present` to tell compositor to continue rendering.
         self.waiting_on_present = true;
-        let msg = ConstellationMsg::ReadyToPresent(
-            self.root_content_pipeline.top_level_browsing_context_id,
-        );
-        if let Err(e) = self.constellation_chan.send(msg) {
-            warn!("Sending event to constellation failed ({:?}).", e);
+        for (&top_level_browsing_context_id, _) in self.webviews.painting_order() {
+            let msg = ConstellationMsg::ReadyToPresent(top_level_browsing_context_id);
```

Comment:
Can the `ReadyToPresent` carry an array now versus sending so many different messages?

Reply:
done in 32aaf283ee1c6dc6d840544d8133dba402ecb552